kernel/arch/riscv/irq.c

```diff
@@ -12,48 +12,127 @@
 #include <tilck/kernel/worker_thread.h>
 #include <tilck/kernel/timer.h>
 
+#include <tilck/mods/irqchip.h>
+
+extern struct irq_data irq_datas[MAX_IRQ_NUM];
+extern ulong irq_bitmap[MAX_IRQ_NUM / (sizeof(ulong) * 8)];
 
 struct list irq_handlers_lists[MAX_IRQ_NUM];
 
 void irq_set_mask(int irq)
 {
-   NOT_IMPLEMENTED();
+   ulong var;
+   u32 hwirq;
+   struct irq_domain *domain;
+   ASSERT(IN_RANGE_INC(irq, 0, MAX_IRQ_NUM - 1));
+
+   disable_interrupts(&var);
+   if (irq_datas[irq].present) {
+
+      domain = irq_datas[irq].domain;
+      hwirq = irq_datas[irq].hwirq;
+      domain->ops->hwirq_set_mask(hwirq, domain->priv);
+   }
+   enable_interrupts(&var);
 }
 
 void irq_clear_mask(int irq)
 {
-   NOT_IMPLEMENTED();
+   ulong var;
+   u32 hwirq;
+   struct irq_domain *domain;
+   ASSERT(IN_RANGE_INC(irq, 0, MAX_IRQ_NUM - 1));
+
+   disable_interrupts(&var);
+   if (irq_datas[irq].present) {
+
```

Comment:
Extra empty line because the following line is longer than the '{' on the IF line.

Reply:
fixed！